### PR TITLE
Batched messages

### PR DESCRIPTION
**Pull request by gjaldon**
This is for applying the batched transactions fix on older connectors that are not yet fully using the connector library. That way, we can apply the producer tx timeout fixes with little or no refactoring.

### CODE REVIEW

File: kafkautils/producer.go
Comment (rukolahasser) [inline kafkautils/producer.go:302]: why do you move it here?

**Review by rukolahasser — COMMENTED** [review]
@cnkarz Could you have a look please? I remember you implemented the same feature.

**Comment by cnkarz** [comment]
- I think reordering as produce-commit-begin is bad practice but it might be necessary in this use-case.
- We have to check all connectors since not all use the `WriteAndCommitSink()` and if they call `WriteAndCommit()` instead, they will have fatal error. We either should fix these connectors or make some methods private (preferably both)
- I would still recommend using `InitProduceChannel()` for future implementations.

**Comment by gjaldon** [comment]
@rukolahasser @cnkarz this does not use `InitProduceChannel()` because that expects the connectors to use the connector library's `Connector` struct. Since there are several connectors already using `WriteAndCommitSink()`, I only wanted to modify that function to include the fix for transaction timeouts without having to rewrite these older connectors. 

If I'm not mistaken, using the Connector struct in the connector library will mean having to rewrite more of the connector and migrating to the connectors repo so it builds the manifest.yml file into the image. Don't want to do all that since I just want to fix transaction timeout.

File: kafkautils/producer.go
Comment (gjaldon) [reply kafkautils/producer.go:302]: It's so I can call `WriteAndCommit()` in a loop. As far as I know, calling `BeginTransaction()` at the start or after did not have an effect on transaction timeouts.

**Comment by gjaldon** [comment]
I will update the 4 connectors in the blep repo that use `WriteAndCommit()` once this PR is merged. I will need to update the connector lib version in there anyway.

**Comment by rukolahasser** [comment]
@cnkarz It's your call. Please note that we discussed refactoring connectors in today's meeting, don't forget to factor it in.

**Review by cnkarz — APPROVED** [review]
I think it is still a necessary quick fix. Maybe we can mark these methods as deprecated so that new devs don't use them.

**Pull request merged**
